include/deal.II/lac/la_vector.templates.h

```diff
@@ -0,0 +1,319 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C)  2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__la_vector_templates_h
+#define dealii__la_vector_templates_h
+
+#include <deal.II/lac/la_vector.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  template <typename Number>
+  VectorSpaceVector<Number> &Vector<Number>::operator*= (const Number factor)
+  {
+    AssertIsFinite(factor);
+    for (unsigned int i=0; i<this->size(); ++i)
+      this->val[i] *= factor;
+
+    return *this;
+  }
+
+
+
+  template <typename Number>
+  VectorSpaceVector<Number> &Vector<Number>::operator/= (const Number factor)
+  {
+    AssertIsFinite(factor);
+    Assert(factor!=Number(0.), ExcZero());
+    this->operator *= (Number(1.)/factor);
+
+    return *this;
+  }
+
+
+
+  template <typename Number>
+  VectorSpaceVector<Number> &Vector<Number>::operator+= (const VectorSpaceVector<Number> &V)
+  {
+    // Downcast V. If fails, throws an exception.
+    const Vector<Number> &down_V = dynamic_cast<const Vector<Number>&>(V);
```

Comment:
(`dynamic_cast` only throws an exception if applied to *references* that cannot be cast. It simply returns a NULL pointer for *pointers*, and this you can check.)